engine/player/covenant.cpp

```diff
@@ -499,4 +517,123 @@ report::sc_html_stream& covenant_state_t::generate_report( report::sc_html_strea
 
   return root;
 }
-} // Namespace covenant ends
+
+void initialize_soulbinds( player_t* player )
+{
+  if ( !player->covenant )
+    return;
+
+  for ( auto soulbind_spell : player->covenant->soulbind_spells() )
+  {
+    auto spell = player->find_spell( soulbind_spell );
+    if ( !spell->ok() )
+      continue;
+
+    special_effect_t effect { player };
+    effect.type = SPECIAL_EFFECT_EQUIP;
+    effect.source = SPECIAL_EFFECT_SOURCE_SOULBIND;
+
+    unique_gear::initialize_special_effect( effect, soulbind_spell );
+
+    player->special_effects.push_back( new special_effect_t( effect ) );
```

Comment:
It might be wise here to ensure that the soulbind special effect is a custom one. We have had issues where the "auto infer" special effect system has tried to be clever and completely messed up for example Legion legendary spells.

So, adding a if ( !effect.is_custom() ) { continue; } here  before we push the special effect copy to player would probably be a good idea.